File: app/src/main/java/com/wojdor/hearthcards/application/view/SummaryValueListPreference.java
package com.wojdor.hearthcards.application.view;

import android.content.Context;
import android.support.v7.preference.ListPreference;
import android.util.AttributeSet;

public class SummaryValueListPreference extends ListPreference {

    public SummaryValueListPreference(Context context) {
        super(context);
    }

    public SummaryValueListPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public CharSequence getSummary() {
        return getEntry();
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/app/view/SummaryValueListPreference.kt
package com.wojdor.hearthstonecards.app.view

import android.content.Context
import android.support.v7.preference.ListPreference
import android.util.AttributeSet

class SummaryValueListPreference : ListPreference {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    override fun getSummary(): CharSequence = entry
}
